rust/chains/hyperlane-ethereum/src/contracts/mailbox.rs

```diff
@@ -328,6 +330,70 @@ where
         };
         fill_tx_gas_params(tx, self.provider.clone(), &tx_overrides).await
     }
+
+    async fn simulate_batch(
+        &self,
+        multicall: &mut Multicall<M>,
+        contract_calls: Vec<ContractCall<M, ()>>,
+    ) -> ChainResult<BatchSimulation<M>> {
+        let batch = multicall::batch::<_, ()>(multicall, contract_calls.clone()).await;
+        let batch = self.add_gas_overrides(batch, None).await?;
```

Comment:
Do we need to call this here? Because we're just calling I think it's fine to skip this, I believe these are mostly in service of prepping a call to be a full on tx. E.g. we don't need the gas price / gas limit set and could cut out the extra RPCs
Feels more appropriate in `submit_tx`